src/time_unit/days_of_month.rs

```diff
@@ -5,7 +5,7 @@ use once_cell::sync::Lazy;
 
 static ALL: Lazy<OrdinalSet> = Lazy::new(|| { DaysOfMonth::supported_ordinals() });
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq)]
```

Comment:
Deriving `PartialEq` means that some equivalent representations of "all" may not be found to be equal. (e.g. `*` and `?` would be considered equal, but `*` and `1-7` for `DayOfWeek` would not.)

We could address this by either:
* Manually implementing `PartialEq` and checking for the `is_all()` case OR
* Adding code to the parsing logic that checks to see if we're about to call `from_ordinal_set()` using an ordinal set that's effectively "all" and using `all()` instead, guaranteeing that the internal representation is the same.

I think I'd prefer the former as it consolidates the definition of `PartialEq` into a single function while the latter relies on knowing about an invariant in the parser. It seems like we could do a single blanket impl of `PartialEq`?

```rust
impl <T> PartialEq for T where T: TimeUnitField { /*....*/ }
```

With that in place, `ScheduleFields` could use `derive(PartialEq, Eq)`.

Reply:
You're right, good catch! I can do with both ideas, so I'll work on the former

Reply:
Sadly it seems that 
```rust
impl <T> PartialEq for T where T: TimeUnitField { /*....*/ }
```
runs into some conflicts. I can go forward and implement it individually though

Reply:
Oof, I've stubbed my toe on that more than once and still forgot about it.